Added new inventory file topics (#555)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/ref-hub-variables.adoc

[id="ref-hub-variables"]

= {HubNameMain} variables

[cols="50%,50%",options="header"]
|====
| *Variable* | *Description* 
| *`automationhub_admin_password`* | Required
| *`automationhub_api_token`* | If upgrading from {PlatformNameShort} 2.0 or earlier, you must either:

- provide an existing {HubNameMain} token as `automationhub_api_token`, or 

- set `generate_automationhub_token` to `true` to generate a new token

Generating a new token will invalidate the existing token.

> | *`automationhub_authentication_backend`* | ldap ?? TBD

Reviewer: It's not set by default and you'll set to 'ldap' to use LDAP authentication. When this is set to ldap, the follwing variables must also be set:

automationhub_ldap_server_uri
automationhub_ldap_bind_dn
automationhub_ldap_bind_password
automationhub_ldap_user_search_base_dn
automationhub_ldap_group_search_base_dn

> Set it to `ldap`` to use LDAP authentication. 

Reviewer 2: Is this a typo: `ldap`` ?

In line 19, it appears as `'ldap'`

Author: No. Text variables should be in quotation marks, so `'ldap'`

Reviewer 2: Sorry, I wasn't clear in my comment. The text variable only shows one quotation mark when rendered.

| *`automationhub_pg_database`* | _Required_

The database name.

The default value is `‘automationhub’`
| *`automationhub_pg_host`* | Required if not using internal database.
| *`automationhub_pg_password`* | The password for the {HubName} postgreSQL database.

Do not use special characters for `automationhub_pg_password`. 
They can cause the password to fail.
| *`automationhub_pg_port`* | Required if not using internal database.

Default value is 5432
| *`automationhub_pg_sslmode`* | Required.

Default value is `‘prefer’`
| *`automationhub_pg_username`* | Required 

Default value is `‘automationhub’`
| *`automationhub_require_content_approval`* | _Optional_

If {HubName} enforces the approval mechanism before collections are made available.

By default when you upload collections to {HubName} an administrator must approve it before it is made available to the users. 

If you want to disable the content approval flow, the variable should be set to `false`.

Default = `true`
| *`automationhub_ssl_cert`* | _Optional_ 

`/path/to/automationhub.cert`
Same as `web_server_ssl_cert` but for {HubName} UI and API
| *`automationhub_ssl_key`* | _Optional_

> `/path/to/automationhub.key``

Author: yes. fixed.

If set to `true` a signing service is created.

Default = `false`
| *`automationhub_disable_hsts`* | The default install deploys a TLS enabled {HubNameMain}.
use if {HubName} is deployed with _HTTP Strict Transport Security_ (HSTS) web-security policy enabled.
Unless specified otherwise, the HSTS web-security policy mechanism is enabled. 
This setting allows you to disable it if required.

Default = `false`
| *`automationhub_disable_https`* | _Optional_

If {HubNameMain} is deployed with HTTPS enabled.

Default = `false`.
| *`automationhub_enable_api_access_log`* | When set to `true`, creates a log file at `/var/log/galaxy_api_access.log` that logs all user actions made to the platform, including their username and IP address.

Default = `false`.
| *`automationhub_importer_settings`* | _Optional_ 

Dictionary of setting to pass to galaxy-importer.

At import time collections can go through a series of checks.

Behavior is driven by `galaxy-importer.cfg` configuration.

Examples are `ansible-doc`, `ansible-lint`, and `flake8`.

This parameter enables you to drive this configuration.

> 2+| For {HubNameMain} to connect to LDAP directly; the following variables must be configured. 

Reviewer 3: Should this go into a separate table?
Accessibility guidelines don't recommend multi-column table cells.

A list of other LDAP related variables (not covered by the `automationhub_ldap_xxx` variables below) that can be passed using the `ldap_extra_settings` variable can be found here:
https://django-auth-ldap.readthedocs.io/en/latest/reference.html#settings
| *`automationhub_ldap_bind_dn`* | The name to use when binding to the LDAP server with `automationhub_ldap_bind_password`. 
| *`automationhub_ldap_bind_password`* | _Required_

The password to use with `automationhub_ldap_bind_dn`.
| *`automationhub_ldap_group_search_base_dn`* | An LDAPSearch object that finds all LDAP groups that users might belong to. 
If your configuration makes any references to LDAP groups, this and `automationhub_ldap_group_type` must be set.

> Default = '`None`'

Reviewer 3: ```suggestion
Default = `'None'`
```

| *`automatiohub_ldap_group_search_filter`* | _Optional_

Search filter for finding group membership.

Default = `'(objectClass=Group)'`
| *`automationhub_ldap_group_search_scope`* | _Optional_

Default = `'SUBTREE'`
| *`automationhub_ldap_group_type_class`* | _Optional_

Default =`'django_auth_ldap.config:GroupOfNamesType'`
| *`automationhub_ldap_server_uri`* | The URI of the LDAP server. 
This can be any URI that is supported by your underlying LDAP libraries.
| *`automationhub_ldap_user_search_base_dn`* | An LDAPSearch object that locates a user in the directory. 
The filter parameter should contain the placeholder %(user)s for the username. 
It must return exactly one result for authentication to succeed.
| *`automationhub_main_url`* | When using Single Sign-On, specify the main {HubName} URL that

> clients will connect to, e.g. `\https://<hubaddress.example.com>`, which leads to the external address being entered in `/etc/pulp/settings.py`.

Reviewer 3: ```suggestion
clients will connect to, for example `\https://<hubaddress.example.com>`, which leads to the external address being entered in `/etc/pulp/settings.py`.
```

If not specified, the first node in the `[automationhub]` group is used.
| *`automationhub_pg_database`* | _Required_

The database name.

Default = `‘automationhub’`
| *`automationhub_pg_host`* | Required if not using internal database.

> | *`automationhub_pg_password`* | The password for the {HubName} postgreSQL database.

Reviewer 3: ```suggestion
| *`automationhub_pg_password`* | The password for the {HubName} PostgreSQL database.
```

Do not use special characters for `automationhub_pg_password`. 
They can cause the password to fail.
| *`automationhub_pg_port`* | Required if not using internal database.

Default = 5432
| *`automationhub_pg_sslmode`* | Required.

Default = `‘prefer’`
| *`automationhub_pg_username`* | Required 

Default = `‘automationhub’`
| *`automationhub_require_content_approval`* | _Optional_

If {HubName} enforces the approval mechanism before collections are made available.

By default when you upload collections to {HubName} an administrator must approve it before it is made available to the users. 

> If you want to disable the content approval flow, the variable should be set to `false`.

Reviewer 3: ```suggestion
If you want to disable the content approval flow, set the value to `false`.
```

Default = `true`
| *`automationhub_ssl_cert`* | _Optional_ 

`/path/to/automationhub.cert`
Same as `web_server_ssl_cert` but for {HubName} UI and API
| *`automationhub_ssl_key`* | _Optional_

`/path/to/automationhub.key`

Same as `web_server_ssl_key` but for {HubName} UI and API
| *`automationhub_ssl_validate_certs`* | For {PlatformName} 2.2 and later, this value is no longer used.

> If {HubName} should validate certificate when requesting itself because by default, Platform deploys with self-signed certificates.

Reviewer 3: ```suggestion
If {HubName} should validate certificate when requesting itself because by default, {PlatformNameShort} deploys with self-signed certificates.
```

Default = `'false'`.
| *`automationhub_upgrade`* | By default, when the {HubName} package and its dependencies are installed they are not upgraded when running the installer, even if newer packages are available. 

To upgrade, you must to run the `./setup.sh` script with this variable set to `true`.

Default = `'false'`
| *`generate_automationhub_token`* a| If upgrading from {PlatformName} 2.0 or earlier, you must either:

* provide an existing {HubNameMain} token as `automationhub_api_token` or

* set `generate_automationhub_token` to `true` to generate a new token.
Generating a new token will invalidate the existing token.

> | *`pulp_db_fields_key`* | Relative or absolute path to the Fernet symmetric encryption key one wants to import. 

Reviewer 3: ```suggestion
| *`pulp_db_fields_key`* | Relative or absolute path to the Fernet symmetric encryption key you want to import. 
```

The default value is `false`.
| *`automationhub_importer_settings`* | _Optional_ 

At import time collections can go through a series of checks.

Behavior is driven by `galaxy-importer.cfg` configuration.

Examples are `ansible-doc`, `ansible-lint`, and `flake8`.

This parameter enables you to drive this configuration.

This variable is expected to be a dictionary.
2+| For {HubNameMain} to connect to LDAP directly the following variables must be configured. 
A list of other LDAP related variables (not covered by the `automationhub_ldap_xxx` variables below) that can be passed using the `ldap_extra_settings` variable can be found here:
https://django-auth-ldap.readthedocs.io/en/latest/reference.html#settings
| *`automationhub_ldap_bind_dn`* | The name to use when binding to the LDAP server with `automationhub_ldap_bind_password`). 

Use the empty string (the default) for an anonymous bind. 

To authenticate a user, bind with that user’s DN and password, but for all other LDAP operations, the user is bound as the DN in this setting. 

The default is '' (Empty string)
| *`automationhub_ldap_bind_password`* | The password to use with `automationhub_ldap_bind_dn`.

> The default is '' (Empty string)

Author: Is there a default value?

> The default is '' (Empty string)

Reviewer: No default value

| *`automationhub_ldap_group_search`* | An LDAPSearch object that finds all LDAP groups that users might belong to. 
If your configuration makes any references to LDAP groups, this and `automationhub_ldap_group_type` must be set.

The default is '`None`'
| *`automatiohub_ldap_group-search_filter`* | _Optional_

Default = `'(objectClass=Group)'`
| *`automationhub_ldap_group_search_scope`* | _Optional_

Default = `'subtree'`

> | *`automationhub_ldap_group_type`* |An LDAPGroupType instance describing the type of group returned by `automationhub_ldap_group_search`.

Reviewer: We don't have this variable.

| *`automationhub_ldap_group_type_class`* | _Optional_

Default =`'django_auth_ldap.config:GroupOfNamesType'`
| *`automationhub_ldap_server_uri`* | The URI of the LDAP server. 

This can be any URI that is supported by your underlying LDAP libraries. 

Can also be a callable that returns the URI. 

The callable is passed a single positional argument: `request`.

When `automationhub_ldap_server_uri` is set to a callable, it is passed a positional request argument. 

Support for no arguments continues for backwards compatibility but will be removed in a future version.
| *`automationhub_ldap_user_search_base_dn`* | An LDAPSearch object that locates a user in the directory. 
The filter parameter should contain the placeholder %(user)s for the username. 
It must return exactly one result for authentication to succeed.

The default is `‘None’`
| *`automationhub_ldap_user_search_filter`* |  _Optional_

> Default = `'subtree'`

Reviewer: `SUBTREE`
